core/src/main/scala/org/apache/spark/internal/config/Network.scala

```diff
@@ -51,6 +51,21 @@ private[spark] object Network {
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString(STORAGE_BLOCKMANAGER_TIMEOUTINTERVAL.defaultValueString)
 
+  private[spark] val NETWORK_EXECUTOR_TIMEOUT =
+    ConfigBuilder("spark.network.executorTimeout")
+      .version("3.4.0")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createOptional
+
+  private[spark] val HEARTBEAT_EXPIRY_CANDIDATES_TIMEOUT =
+    ConfigBuilder("spark.network.expiryCandidatesTimeout")
+      .doc("This config is a timeout used for heartbeat receiver `executorExpiryCandidates`. Be" +
+        "effective only when HEARTBEAT_RECEIVER_CHECK_WORKER_LAST_HEARTBEAT is enabled. See" +
+        "[SPARK-39984] for more details")
+      .version("3.4.0")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("30s")
```

Comment:
We should namespace these configs to heartbeat, thoughts @Ngone51  ?

Reply:
+1. 

We can add this config to the namespace `spark.executor.heartbeat`. cc @kevin85421

Reply:
Updated https://github.com/apache/spark/pull/37411/commits/e14853527434b7d625f0de05800314d00aaa5fc2